[IGNORE] Add doc about secret (#1471)
perses/perses: docs/secret.md

# Secret

> When you are defining a datasource, likely you will need to provide a basic authentication, a certificate, or a token

Reviewer: ```suggestion
When defining a datasource, you will probably need to provide a basic authentication, a certificate, or a token
```

to be used when the Perses backend will contact your Datasource.

`Secret` or `GlobalSecret` are the two objects that you need. The choice between a `Secret` or

> a `GlobalSecret` will mainly depend on if you are defining a `Datasource` or a `GlobalDatasource`.

Reviewer: Maybe we should highlight more the difference in terms of "scope behavior" with the rest of the resources here?
I mean e.g a project variable can rely on a global datasource for instance, while here we have:
- GlobalDatasource -> GlobalSecret
- (Project) Datasource -> Secret only
- Dashboard datasource -> didnt check but Secret only too I guess ?

Maybe it's an issue to have such discrepancy btw, not sure but anyway that's out of the topic of this PR.

WDYT?

Author: ah yeah indeed it's certainly unclear like that. I will change that

# Secret

When defining a datasource, you will probably need to provide a basic authentication, a certificate, or a token to be
used when the Perses backend will contact your Datasource.

We have two different objects to store sensitive data: `GlobalSecret` and `Secret`.

> To know what which one you need, it will depend on what object representing a datasource you are using.

Reviewer: ```suggestion
You should use one or the other depending on which object your datasource corresponds to.
```

> If you want to store sensitive data for a `GlobalDatasource`, then you need to create a `GlobalSecret`.

Reviewer: ```suggestion
- To store sensitive data for a `GlobalDatasource`, you need to create a `GlobalSecret`.
- For a `Datasource` object or a datasource defined directly in a dashboard , you need to create a `Secret`. `GlobalSecret` cannot be used here.
```
